### PR TITLE
[WIP] Module rework

### PR DESCRIPTION
**Pull request by lolleko**

### CODE REVIEW

File: src/Transpiler.ts [hunk 1]
```diff
@@ -49,6 +65,22 @@ export class LuaTranspiler {
...
     }
 
+    accessPrefix(node?: ts.Node): string {
+        return node && this.isModule ?
+            "local " : ""
+    }
+
+    makeExport(name: string | ts.__String, node: ts.Node): string {
+        let result: string = "";
+        if (node && node.modifiers && !!(ts.getCombinedModifierFlags(node) & ts.ModifierFlags.Export)) {
```
Comment (Perryvw) [inline src/Transpiler.ts]: Another `!!`, hacky

File: src/Transpiler.ts [hunk 2]
```diff
@@ -751,6 +802,9 @@ export class LuaTranspiler {
         // Close function block
         result += this.indent + "end\n";
 
+        result += this.makeExport(methodName, node);
+
+
```
Comment (Perryvw) [inline src/Transpiler.ts]: too many newlines

File: test/integration/lua/modules.spec.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        `local test0 = require("test")
+        local RenamedClass = test0.TestClass`
+    )
+    @TestCase(
+        `namespace TestSpace {}`,
+        `TestSpace = TestSpace or {}
+        exports.TestSpace = exports.TestSpace or {}
+        do
+        end`
+    )
```
Comment (Perryvw) [inline test/integration/lua/modules.spec.ts]: Add test with non-empty namespace.

Comment (Perryvw) [inline test/integration/lua/modules.spec.ts]: Add test with (non-empty) class.

File: test/integration/lua/modules.spec.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import { Expect, Test, TestCase } from "alsatian";
+import * as util from "../../src/util"
+
+const dedent = require('dedent')
+
+export class LuaLoopTests {
```
Comment (Perryvw) [inline test/integration/lua/modules.spec.ts]: Loop tests?
